src/psaform/psaform.jsx

```diff
@@ -0,0 +1,190 @@
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import { Grid, Typography, Snackbar, Checkbox, FormGroup, FormControlLabel } from "@mui/material";
+import PSAButton from "../button";
+import PSATextField from "../Textfield";
+
+export const PSAForm = ({
+  apiUrl,
+  submitMessage,
+  headerTitle,
+  fields, // Single prop for both text fields and checkboxes
+  buttons,
+  consentRedux,
+  pirschAnalytics,
+}) => {
+  const [isSubmitDisabled, setIsSubmitDisabled] = useState(true);
+
+  const remove = (arr, value) => arr.filter((item) => item !== value);
```

Comment:
Can you put this function into `handleCheckboxChange()` since it's only being used there